sync/sync-store/src/main/java/com/duckduckgo/sync/store/SyncDatabase.kt

```diff
@@ -51,8 +56,21 @@ object SyncTypeConverters {
     }
 
     @TypeConverter
-    @JvmStatic
     fun fromSyncState(syncState: SyncAttemptState): String {
         return syncState.name
     }
+
+    @TypeConverter
+    fun toSyncApiErrorType(errorType: String): SyncApiErrorType {
+        return try {
+            SyncApiErrorType.valueOf(errorType)
+        } catch (t: Throwable) {
+            SyncApiErrorType.TOO_MANY_REQUESTS
```

Comment:
Won't be true to signal a too many request here. what about having `unknown` as fallback type? and if `unknown`, we don't send a pixel.

Reply:
actually not, because the chain goes all the way. the right thing to do here is to throw IllegalArgumentException

Reply:
That's correct. However, be careful if we ever remove an errorType or the db will crash when reading an old value.